File: cpp/example/src/ContainerWithMostH2O/containerWithMostH2O.cpp
#include "containerWithMostH2O.h"

int containerWithMostH2O::naive(vector<int>& height){
    int l=0;int r=height.size()-1;
    int maxV = min(height[l],height[r])*r;
    while (l<r){
        if (height[l]>=height[r])r-=1;
        else l+=1;
        maxV = max(maxV,min(height[l],height[r])*(r-l));
    }
    return maxV;
}

File: cs/sample/Sample/ContainerWithMostH2O/ContainerWithMostH2O.cs
using System;
using System.Collections.Generic;

namespace Sample.ContainerWithMostH2O{
    public class containerWithMostH2O{
        public static int naive(int[] height){
            int l=0;int r=height.Length-1;
            int maxV = Math.Min(height[l],height[r])*r;
            while (l<r){
                if (height[l]>=height[r]) r-=1;
                else l+=1;
                maxV = Math.Max(maxV,Math.Min(height[l],height[r])*(r-l));
            }
            return maxV;
        }
    }
}
